src/pybind/mgr/dashboard/frontend/src/app/shared/forms/cd-validators.spec.ts

```diff
@@ -201,6 +201,29 @@ describe('CdValidators', () => {
       expect(x.hasError('match')).toBeFalsy();
       expect(y.hasError('match')).toBeFalsy();
     });
+
+    it('should unset error when values are equal', () => {
+      const x = form.get('x');
+      const y = form.get('y');
+      x.setValue('aaa');
+      y.setValue('aaa');
+      y.setErrors({ match: true });
+      CdValidators.match('x', 'y')(form);
+      expect(x.hasError('match')).toBeFalsy();
+      expect(y.hasError('match')).toBeFalsy();
```

Comment:
You should also expect if `y` is valid now, to test the case you are describing in the comment.